base/compiler/abstractinterpretation.jl

```diff
@@ -689,6 +689,10 @@ function edge_matches_sv(interp::AbstractInterpreter, frame::AbsIntState,
     if callee_method2 !== inf_method2
         return false
     end
+    if isa(frame, InferenceState) && frame.interp !== interp
```

Comment:
The isa check makes it seem like this is doing something incorrectly. Isn't this actually the `is_same_frame` query though on the line above here?

Anyways, SGTM as a hack, but I dislike that the `NativeInterpreter` seems like a gigantic stack object that now needs to be memcmp on every frame against every other frame here.

Reply:
It's kind of is_same_frame, but on the method. Rather than comparing the interpreter, I think we could compare the `cache_owner` if you like that better.

Reply:
yeah, `cache_owner` does intuitively feel like the right level of abstraction to me